examples/general_demo.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+""" This script will:
+- download unsorted dicoms
+- run dcm2bids to convert to nifti with bids structure
+- unwrap phase
+- save wrapped and unwrapped plot of first X,Y volume as unwrap_phase_plot.png in the output directory
+"""
+
+import os
+import numpy as np
+from matplotlib.figure import Figure
+import glob
+import logging
+
+import nibabel as nib
+
+from shimmingtoolbox.unwrap import prelude
+from shimmingtoolbox.utils import run_subprocess
+from shimmingtoolbox import __dir_testing__
+from shimmingtoolbox import dicom_to_nifti
+
+
+def general_demo(path_output=os.path.join(os.path.curdir, 'output_dir')):
+    """
+    Args:
+        path_output (str): Output directory to store data and results.
+
+    Returns:
+        str: File name and path of output figure
+    """
+
+    logging.basicConfig(level='INFO')
+
+    # Download example data
+    path_testing_data = os.path.join(path_output, __dir_testing__)
+    run_subprocess('st_download_data {} --output {}'.format(__dir_testing__, path_testing_data))
+
+    # Transfer from dicom to nifti
+    path_dicom_unsorted = os.path.join(path_testing_data, 'dicom_unsorted')
+    path_nifti = os.path.join(path_output, 'niftis')
+    dicom_to_nifti(path_dicom_unsorted, path_nifti, subject_id='sub-01')
+
+    # Open phase data
+    fname_phases = glob.glob(os.path.join(path_nifti, 'sub-01', 'fmap', '*phase*.nii.gz'))
+
+    nii_phase_e1 = nib.load(fname_phases[0])
+    nii_phase_e2 = nib.load(fname_phases[1])
+
+    # Scale phase to radians
+    phase_e1 = np.interp(nii_phase_e1.get_fdata(), [0, 4096], [-np.pi, np.pi])
+    phase_e2 = np.interp(nii_phase_e2.get_fdata(), [0, 4096], [-np.pi, np.pi])
```

Comment:
When scaling to radians, are each of these args always the same except for the first? If so, it might benefit maintainability to abstract this scaling operation into a helper function that can be imported since it's repeated here and in `test_prelude.py`. Perhaps it could go in `utils.py` and be imported from there.

Reply:
@gab-berest also had this idea, my initial thought is that `[0, 4096]` might change for non siemens data. We only deal with siemens so it might not be relevant but I still like the idea of being flexible and not introduce a source of error. So now with 2 input arguments I'm not sure it is worth it to introduce a new `scale_phase(data, input_range)` if people are already familiar with interp. What do you think?

Reply:
>  I'm not sure it is worth it to introduce a new scale_phase(data, input_range)

sounds like the best option to me: isolating the "scaling" problem, with unknown data provenance (e.g. from siemens, philips, ge, toshiba, different sequences, software versions, etc.). And you could create an exhaustive sample of data and a specific unit test.

Reply:
Opened an [issue](https://github.com/shimming-toolbox/shimming-toolbox-py/issues/64) to implement a scale_phase function. It is out of scope of this PR so it will be implemented in the future.